x-pack/test/functional/apps/lens/drag_and_drop.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['visualize', 'lens', 'common', 'header']);
+
+  describe('lens drag and drop tests', () => {
+    it('should construct the basic split xy chart', async () => {
+      await PageObjects.visualize.navigateToNewVisualization();
+      await PageObjects.visualize.clickVisType('lens');
+      await PageObjects.lens.goToTimeRange();
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await PageObjects.lens.dragFieldToWorkspace('@timestamp');
+
+      expect(await PageObjects.lens.getDimensionTriggerText('lnsXY_xDimensionPanel')).to.eql(
+        '@timestamp'
+      );
+    });
+
+    it('should allow dropping fields to existing and empty dimension triggers', async () => {
+      await PageObjects.lens.switchToVisualization('lnsDatatable');
+
+      await PageObjects.lens.dragFieldToDimensionTrigger(
+        'clientip',
+        'lnsDatatable_column > lns-dimensionTrigger'
+      );
+      expect(await PageObjects.lens.getDimensionTriggerText('lnsDatatable_column')).to.eql(
+        'Top values of clientip'
+      );
+
+      await PageObjects.lens.dragFieldToDimensionTrigger(
+        'bytes',
+        'lnsDatatable_column > lns-empty-dimension'
+      );
+      expect(await PageObjects.lens.getDimensionTriggerText('lnsDatatable_column', 1)).to.eql(
+        'bytes'
+      );
+      await PageObjects.lens.dragFieldToDimensionTrigger(
+        '@message.raw',
+        'lnsDatatable_column > lns-empty-dimension'
+      );
+      expect(await PageObjects.lens.getDimensionTriggerText('lnsDatatable_column', 2)).to.eql(
+        'Top values of @message.raw'
+      );
+    });
+
+    it('should reorder the elements for the table', async () => {
+      await PageObjects.lens.reorderDimensions('lnsDatatable_column', 2, 0);
+      expect(await PageObjects.lens.getDimensionTriggersTexts('lnsDatatable_column')).to.eql([
```

Comment:
@wylieconlon, not adding the waiting for visualisation (or more precisely, waiting for an update in UI after finishing drag and drop action) causes this to fail. This is happening for the following reason:
- selenium drags and drops an element to reorder, but the UI (dimension triggers) didn't update yet
- the expect function is being executed - it immediately finds elements it looks for, so it doesn't have to wait for UI to update. 
- UI updates

I don't want to start reverting your commits while you're reverting mines, so let's talk during sync how you want to handle this.

https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/964/

Reply:
Let's bring back the waiting to resolve this issue.